packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -99,6 +106,24 @@ export default class CodeMode extends Component<Signature> {
     }
   });
 
+  @use importedModule = resource(() => {
+    if (isReady(this.openFile.current)) {
+      let f: Ready = this.openFile.current;
+      if (f.name.endsWith('.json')) {
+        let ref = identifyCard(this.cardResource.value?.constructor);
+        if (ref !== undefined) {
+          return importResource(this, () => moduleFrom(ref as CodeRef));
+        } else {
+          return;
+        }
+      } else {
+        return importResource(this, () => f.url);
+      }
+    } else {
+      return undefined;
+    }
+  });
```

Comment:
added new import here which uses the previous importModule resource that was used by the schema editor. When the url is `author.gts` it loads the url from the file resource. When the url is `Author/1.json` it loads the module url (not the instance one)

Reply:
I did it differently previously but I figured this is the way its supposed to be designed since other components like the schema editor will consume from this resource.